egg/core/trainers.py

```diff
@@ -304,3 +306,16 @@ def load_from_latest(self, path):
 
         if latest_file is not None:
             self.load_from_checkpoint(latest_file)
+
+
+def extract_kwargs(batch):
+    """
+    Check whenever the batch format is (args, kwargs) and return proper split
+    """
+    if isinstance(batch[1], dict):
+        args, kwargs = batch
+    else:
+        args = batch
+        kwargs = {}
+
+    return args, kwargs
```

Comment:
I think it'd be better to have kwargs by default in every game instead of having this check here. It looks like a patchy/hacky solution

Reply:
I had to implement it far backward compatibility, rather than change every game in zoo

Reply:
I meant every game in core. That's SenderReceiverRNNReinforce, SymbolGameReinforce etc. I believe there are 4 type of games in total in EGG